### PR TITLE
prefer using systemd for oom monitoring

### PR DESCRIPTION
**Pull request by sjenning**

### CODE REVIEW

**Comment by vishh** [comment]
ok to test

**Comment by vishh** [comment]
This change LGTM! Can you clarify the rationale for this change in this PR?

**Comment by sjenning** [comment]
While I'm looking to add log rotation support (#1248) to the path that watches log files, preferring systemd bypasses the issue entirely by reading from journalctl if it is available.  Some systems will have the log files AND systemd and in that case, we want to read from the journal since it hides the log rotation issue.

File: utils/oomparser/oomparser.go [hunk 1]
```diff
@@ -205,13 +205,17 @@ func getSystemFile() (string, error) {
 // initializes an OomParser object and calls getSystemFile to set the systemFile
 // attribute.  Returns and OomParser object and an error
 func New() (*OomParser, error) {
+	parser, err := trySystemd()
+	if err == nil {
+		return parser, err
```
Comment (timstclair) [inline utils/oomparser/oomparser.go]: (nit) return `nil` instead `err` for clarity

**Comment by vishh** [comment]
Given that there are still other distros that are not running systemd yet, it will still be helpful to fix the log rotation issue.

**Comment by sjenning** [comment]
@timstclair updated

**Comment by timstclair** [comment]
Thanks, LGTM

**Pull request merged**